src/core/search/base.h

```diff
@@ -15,9 +17,20 @@ using DocId = uint32_t;
 using FtVector = std::vector<float>;
 
 // Query params represent named parameters for queries supplied via PARAMS.
-// Currently its only a placeholder to pass the vector to KNN.
-struct QueryParams {
-  FtVector knn_vec;
+struct QueryParams : private absl::flat_hash_map<std::string, std::string> {
```

Comment:
Why inherit privately instead of having a member?

Reply:
Because it carries out only one function, wrapping a hash map 🤷🏻‍♂️ 
Will change